### PR TITLE
We need to stop relying on comm=None

### PR DESCRIPTION
**Pull request by JDBetteridge**
# Description

### CODE REVIEW

**Comment by pefarrell** [comment]
I think for user-facing things it makes sense to default comm to COMM_WORLD. But I think that for developer-facing things (like `load`) we should demand that a comm be passed.

**Comment by JDBetteridge** [comment]
I agree, to clarify my comment a bit: I think we should stop relying on the `comm=None` defaulting to `COMM_WORLD` internally. I don't have plans to force users to have to specify an MPI communicator for all of their scripts. However, internally relying on this behaviour makes it hard to track down parallel bugs!

**Review by connorjward — APPROVED** [review]
LGTM

File: firedrake/preconditioners/fdm.py [hunk 1]
```diff
@@ -1594,9 +1595,9 @@ def tabulate_exterior_derivative(Vc, Vf, cbcs=[], fbcs=[], comm=None):
...
     zero = PETSc.Mat()
-    A00 = petsc_sparse(evaluate_dual(c0, f0), comm=PETSc.COMM_SELF) if f0 else zero
-    A11 = petsc_sparse(evaluate_dual(c1, f1), comm=PETSc.COMM_SELF) if c1 else zero
-    A10 = petsc_sparse(evaluate_dual(c0, f1, "grad"), comm=PETSc.COMM_SELF)
+    A00 = petsc_sparse(evaluate_dual(c0, f0), comm=COMM_SELF) if f0 else zero
+    A11 = petsc_sparse(evaluate_dual(c1, f1), comm=COMM_SELF) if c1 else zero
+    A10 = petsc_sparse(evaluate_dual(c0, f1, "grad"), comm=COMM_SELF)
     Dhat = block_mat(diff_blocks(tdim, ec.formdegree, A00, A11, A10), destroy_blocks=True)
     A00.destroy()
     A11.destroy()
```
Comment (wence-) [inline firedrake/preconditioners/fdm.py:1603]: This looks sus if `A00` and/or `A11` are `zero` (which has not been `create()`ed)

Comment (JDBetteridge) [reply firedrake/preconditioners/fdm.py:1603]: I will open an issue and address this in another PR, today I want to try and get CI running

**Pull request merged**